assets/src/edit-story/components/panels/stylePreset/presets.js

```diff
@@ -0,0 +1,279 @@
+  const [activeStyleIndex, setActiveStyleIndex] = useState(null);
+  const colorsRef = useRef(null);
+  const stylesRef = useRef(null);
+  const { fillColors, textColors, textStyles } = stylePresets;
+
+  const getIndexDiff = (key, rowLength) => {
+    switch (key) {
+      case 'ArrowUp':
+        return -rowLength;
+      case 'ArrowDown':
+        return COLORS_PER_ROW;
+      case 'ArrowLeft':
+        return -1;
+      case 'ArrowRight':
+        return 1;
+      default:
+        return 0;
+    }
+  };
+
+  useLayoutEffect(() => {
+    if (colorsRef.current && null !== activeColorIndex) {
+      colorsRef.current.querySelector('[tabindex="0"]').focus();
+    }
+  }, [activeColorIndex]);
+
+  useLayoutEffect(() => {
+    if (stylesRef.current && null !== activeStyleIndex) {
+      stylesRef.current.querySelector('[tabindex="0"]').focus();
+    }
+  }, [activeStyleIndex]);
+
+  useKeyDownEffect(
+    colorsRef,
+    { key: ['up', 'down', 'left', 'right'], shift: true },
+    ({ key }) => {
+      const maxIndex = isText ? textColors.length - 1 : fillColors.length - 1;
+      if (colorsRef.current) {
+        const diff = getIndexDiff(key, COLORS_PER_ROW);
+        const newIndex = Math.max(
+          0,
+          Math.min(maxIndex, (activeColorIndex ?? 0) + diff)
+        );
+        setActiveColorIndex(newIndex);
+      }
+    },
+    [activeColorIndex]
+  );
+
+  useKeyDownEffect(
+    stylesRef,
+    { key: ['up', 'down', 'left', 'right'], shift: true },
+    ({ key }) => {
+      if (stylesRef.current) {
+        const diff = getIndexDiff(key, STYLES_PER_ROW);
+        const newIndex = Math.max(
+          0,
+          Math.min(textStyles.length - 1, (activeStyleIndex ?? 0) + diff)
+        );
+        setActiveStyleIndex(newIndex);
+      }
+    },
+    [activeStyleIndex]
+  );
+
+  const getStylePresetText = (preset) => {
+    const isHighLight =
+      preset.backgroundTextMode === BACKGROUND_TEXT_MODE.HIGHLIGHT;
+    // @todo Confirm text content usage.
+    const text = textContent.substring(0, 8) + '...';
+    return (
+      <TextWrapper>
+        {isHighLight ? (
+          <HighLight background={preset.backgroundColor}>{text}</HighLight>
+        ) : (
+          text
+        )}
+      </TextWrapper>
+    );
+  };
+
+  const colorPresets = isText ? textColors : fillColors;
+  const hasColorPresets = colorPresets.length > 0;
+
+  const groupLabel = isText
+    ? __('Text colors', 'web-stories')
+    : __('Colors', 'web-stories');
+  return (
+    <PanelContent isPrimary padding={hasColorPresets ? null : '0'}>
+      {hasColorPresets && (
+        <>
+          <PresetGroupLabel>{groupLabel}</PresetGroupLabel>
+          <PresetGroup ref={colorsRef}>
+            {colorPresets.map((color, i) => (
+              <ButtonWrapper key={`color-${i}`}>
+                <Color
+                  tabIndex={
+                    activeColorIndex === i || (!activeColorIndex && i === 0)
```

Comment:
Can't we properly initialize `activeColorIndex` to `0` to avoid this `||`?

Reply:
This was also helpful for not setting the focus on the first render, however, I suppose I could use a ref for that.

Reply:
Changed within 76ad91dd